packages/flutter_driver/lib/src/extension/extension.dart

```diff
@@ -371,27 +371,32 @@ class FlutterDriverExtension {
     return null;
   }
 
-  /// Returns a future that waits until no pending frame is scheduled (frame is synced).
+  /// Returns a future that waits until the app is idle.
   ///
   /// Specifically, it checks:
-  /// * Whether the count of transient callbacks is zero.
-  /// * Whether there's no pending request for scheduling a new frame.
+  /// * Whether there's no pending [MethodChannel.invokeMethod] calls.
+  /// * Whether the count of transient callbacks is zero, if "frame sync" is enabled.
+  /// * Whether there's no pending request for scheduling a new frame, if "frame sync" is enabled.
   ///
-  /// We consider the frame is synced when both conditions are met.
+  /// We consider the app idle when the conditions listed above are met.
   ///
   /// This method relies on a Flutter Driver mechanism called "frame sync",
   /// which waits for transient animations to finish. Persistent animations will
   /// cause this to wait forever.
   ///
   /// If a test needs to interact with the app while animations are running, it
-  /// should avoid this method and instead disable the frame sync using
-  /// `set_frame_sync` method. See [FlutterDriver.runUnsynchronized] for more
-  /// details on how to do this. Note, disabling frame sync will require the
-  /// test author to use some other method to avoid flakiness.
-  Future<Result> _waitUntilNoPendingFrame(Command command) async {
-    await _waitUntilFrame(() {
-      return SchedulerBinding.instance.transientCallbackCount == 0
-          && !SchedulerBinding.instance.hasScheduledFrame;
+  /// should instead disable the frame sync using the `set_frame_sync` method.
+  /// See [FlutterDriver.runUnsynchronized] for more details on how to do this.
+  /// Note, disabling frame sync will require the test author to use some other method to avoid flakiness.
```

Comment:
The doc comment will not appear in our API docs since the method is private. Maybe add the comment to the corresponding method in `FlutterDriver`?

Reply:
Ya... if you think it makes sense to expose this API on Driver, I can add it in a follow-up PR.

Reply:
Why not in this one?

Reply:
My main goal for this PR is to make Espresso tests more reliable by calling this waitUntilIdle API.

As you noted in other comments, I think it makes sense to sort out the relationship between this one and the waitUntilNoTransientCallbacks before exposing it to Flutter Driver users... And I suspect it's going to take time and could be a breaking change. I don't want to include all the changes in one PR, considering I'm not familiar with Flutter's release process (e.g. how to handle breaking changes). At the same time, it doesn't block Espresso.

Hope this works for you.

Reply:
@tvolkert + Todd on this thread. Let me know what would be the best way to move forward.